transport.go

```diff
@@ -322,7 +322,7 @@ func (t *Transport) close(e error) {
 		t.handlerMap.Close(e)
 	}
 	if t.server != nil {
-		t.server.setCloseError(e)
+		t.server.close(e, false)
```

Comment:
Why do we need this? isn't this method called from `server.close`?

Reply:
Ignore this. I didn't realise this is being called from transport.close and not from transport.closeServer